### PR TITLE
Introduce grid engine autoscaler package

### LINKED ISSUE #2498: Extract grid engine autoscaler script to a separate module
**Background**

Once #2492 is merged `workflows/pipe-common/scripts/autoscale_sge.py` script will become almost 2k lines. The script consists of a lot of classes with different responsibilities such as grid engine management, autoscaling instances selection, scaling up/down handling, autoscaling cycling, faults recovering and testing.

**Approach**

It seems that the autoscaling script became pretty complex and should be managed as a separate application in cloud pipeline similarly to pipe cli, pipe fuse, fs browser and etc.

Once grid engine autoscaler is a separate application its classes can be split into multiple files and built and distributed as a single python library.

### PR DESCRIPTION
**Pull request by tcibinan**
Resolves #2498.

The pull request extracts grid engine autoscaler `pipeline.hpc` package from original `autoscale_sge.py` script.

### CODE REVIEW

File: workflows/pipe-common/pipeline/hpc/ssh.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#  Copyright 2017-2023 EPAM Systems, Inc. (https://www.epam.com/)
```
Comment (SilinPavel) [inline workflows/pipe-common/pipeline/hpc/ssh.py]: why this file has a name ssh? maybe cmd will be more accurate?

File: workflows/pipe-common/pipeline/hpc/avail.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#  Copyright 2017-2023 EPAM Systems, Inc. (https://www.epam.com/)
```
Comment (SilinPavel) [inline workflows/pipe-common/pipeline/hpc/avail.py]: I think it is better to emphasize that it is about instance, so how about to name this file as `instance_avail`

**Review by SilinPavel — CHANGES_REQUESTED** [review]

File: workflows/pipe-common/pipeline/hpc/ssh.py (hunk 1 above)
Comment (tcibinan) [reply workflows/pipe-common/pipeline/hpc/ssh.py]: Done

File: workflows/pipe-common/pipeline/hpc/avail.py (hunk 2 above)
Comment (tcibinan) [reply workflows/pipe-common/pipeline/hpc/avail.py]: I've extracted `instance` package with `avail`, `provider` and `selector` subpackages.

Comment (SilinPavel) [reply workflows/pipe-common/pipeline/hpc/avail.py]: LGTM

**Review by SilinPavel — APPROVED** [review]

**Pull request merged**